src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -0,0 +1,216 @@
+
+    assertEqualsCourseArrays(courses, expectedCourses);
+  }
+
+  @Test
+  /* Ensure the whole doGet process works as expected given no search filters. */
+  public void GetCourses_NoParamsSet() {
+
+    request = createRequest(request, "", "", "select", "");
+    List<Course> expectedCourses = new ArrayList<>();
+
+    addCourseEntity("CS 105", "Smith", "Spring 2020", 1);
+    expectedCourses.add(new Course("CS 105", "Smith", Long.valueOf(1), "Spring 2020"));
+
+    addCourseEntity("CS 105", "Smith", "Spring 2020", 2);
+    expectedCourses.add(new Course("CS 105", "Smith", Long.valueOf(2), "Spring 2020"));
+
+    addCourseEntity("CS 106", "Smith", "Spring 2020", 3);
+    expectedCourses.add(new Course("CS 106", "Smith", Long.valueOf(3), "Spring 2020"));
+
+    List<Filter> filters = searchObject.getFilters(request); // Tested in makeFiltersAllParams.
+    List<Entity> results = searchObject.getResults(filters); // Tested in getResultsAllFilters.
+    List<Course> courses = searchObject.getCourses(results);
+
+    assertEqualsCourseArrays(courses, expectedCourses);
+  }
+
+  private void addCourseEntityWithList(
+      String name, String professor, String term, int units, List<Entity> entities, boolean toAdd) {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    Entity entity = new Entity("Course");
+    entity.setProperty("name", name);
+    entity.setProperty("professor", professor);
+    entity.setProperty("units", units);
+    entity.setProperty("term", term);
+    db.put(entity);
+    if (toAdd) {
+      entities.add(entity);
+    }
+  }
+
+  private void addCourseEntity(String name, String professor, String term, int units) {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    Entity entity = new Entity("Course");
+    entity.setProperty("name", name);
+    entity.setProperty("professor", professor);
+    entity.setProperty("units", units);
+    entity.setProperty("term", term);
+    db.put(entity);
+  }
+
+  private HttpServletRequest createRequest(
+      HttpServletRequest request, String name, String professor, String term, String units) {
+    request = Mockito.mock(HttpServletRequest.class);
+    when(request.getParameter("courseName")).thenReturn(name);
+    when(request.getParameter("profName")).thenReturn(professor);
+    when(request.getParameter("term")).thenReturn(term);
+    when(request.getParameter("units")).thenReturn(units);
+    return request;
+  }
+
+  private void assertEqualsCourseArrays(List<Course> courses, List<Course> expectedCourses) {
+    assertEquals(courses.size(), expectedCourses.size());
+    for (int i = 0; i < courses.size(); i++) {
+      Course actual = courses.get(i);
+      Course expected = expectedCourses.get(i);
+      assertEquals(actual.name, expected.name);
+      assertEquals(actual.professor, expected.professor);
+      assertEquals(actual.units, expected.units);
+      assertEquals(actual.term, expected.term);
```

Comment:
Why are these actual.term? Is actual the Course object we are not using anymore? (or are you still using it for your own code)

Reply:
I'm still using the Course object, so I don't have to change as much of my code (and it helps simplify it a tiny bit for me)